### DISCUSSION TITLE
Who is HarukaMa?

### DISCUSSION THREAD

**Discussion by fdciabdul** [community/community#23470] [New to GitHub] [answered]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2cc2d369d4abc18f7e9369023a7647d7af913d6c.png" data-download-href="/uploads/short-url/6nYq6Nri5RNnrc5DqDznCRF1XDC.png?dl=1" title="Screenshot_2020_0607_181531" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/31664438/181077250-b4b66fd4-7bba-425a-ba33-5ef533391cc4.png" alt="Screenshot_2020_0607_181531" data-base62-sha1="6nYq6Nri5RNnrc5DqDznCRF1XDC" width="580" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2cc2d369d4abc18f7e9369023a7647d7af913d6c_2_580x500.png, https://aws1.discourse-cdn.com/github/original/2X/2/2cc2d369d4abc18f7e9369023a7647d7af913d6c.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2cc2d369d4abc18f7e9369023a7647d7af913d6c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2cc2d369d4abc18f7e9369023a7647d7af913d6c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_2020_0607_181531</span><span class="informations">699×602 30.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
i wonder , who is HarukaMa itself … he is a bot ? and why her name showing in my repo

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/fdciabdul">@fdciabdul</a>!</p>
<p>We’ll need to know a little more information to dig into this. It says that the latest commit was made by HarukaMa. Is this commit one that you recognize? If so, is it one that you made or does it relate to a pull request that you merged?</p>
<p>Thanks!</p>

**Comment by fdciabdul** [comment]
<p>yez , the latest commit it says by HarukaMa not myself …   and only me who deploy my project to github itself …</p>
<p>and i use git cli to deploy it , not using third party application or something else</p>

**Comment by ThirdEyeRose** [comment]
<p>I know that the latest commit says that it’s by HarukaMa but when you look at its contents, I’m wondering if you recognize it.</p>
<p>My guess is that this commit was signed by an email address that has been claimed by a different user.</p>

**Comment by phatblinkie** [comment]
<p>i have the same problem, this harukama should be banned if they are talking others emails. this cant be right. every time i make a commit, it shows this stranger committed it.</p>

**Reply by HarukaMa** [reply]
You can't blame others when you don't have correct email addresses on your git commits.

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="phatblinkie" data-post="6" data-topic="117042">
<div class="title">
<div class="quote-controls"></div>
 phatblinkie:</div>
<blockquote>
<p>every time i make a commit, it shows this stranger committed it.</p>
</blockquote>
</aside>
<p>Check the author and committer email address recorded in the commit, e.g. using <code>git log --format=fuller</code>. Are both of those yours?</p>

**Comment by yuantuo666** [comment]
I meet the same problem, i think it is because the default email adress of vultr is root@vultr.guest
and @HarukaMa claimed this email address
```
[root@vultr dir]# git log --format=fuller
commit ****
Author:     root <root@vultr.guest>
AuthorDate: Fri May 19 21:40:57 2023 +0800
Commit:     root <root@vultr.guest>
CommitDate: Fri May 19 21:40:57 2023 +0800

    add language and notice set

```
![image](https://github.com/community/community/assets/44130602/8d780a04-e39b-4427-9281-40a88715de72)

**Reply by HarukaMa** [reply]
Yep, that's correct: I have the email address root@vultr.guest on my account, and although it can't be verified for an obvious reason, GitHub would happily link all commits with that email address to my account.

Always check the email address when you commit directly from servers as chances are they have incorrect git configs.

**Comment by HarukaMa** [comment, marked as answer]
IMO the most obvious fix is make GitHub stop linking commits to unverified email addresses. Or GitHub can just reject non-existent domains (until .guest becomes a real TLD). Or just disallow this one email address to be added to accounts.

I'm not removing that email address from my account. Either use a correct email address in your commits, or make GitHub change how account linking works.

I'll add a notice in my bio to explain this to other "victims".

**Reply by mieko** [reply]
This position makes total sense.  Thanks for the bio notice.

**Reply by fdciabdul** [reply]
its been 3 years , i understand yet , that's why i see your name in there :)

**Comment by nadirhamid** [comment]
I have noticed this issue also. It is simple to correct, although it can also be overlooked.

I may have used a Vultr compute instance and pushed some commits from it, which got linked to @HarukaMa account. 

I certainly think Github can fix this issue by not including the unverified emails in the commit history. I do not want to include @HarukaMa in my project's contributors if they personally did not have any affiliation with me.

This is somewhat of a inconvenience right now. Moreover, removing these commits from Git history is largely complex and not doable if the history is very large.

It would be best if there was a fix implemented for this.

In the meantime, I will be more cautious about how I push to Github. I will verify the email in advance.
